sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveClientImpl.scala

```diff
@@ -332,14 +332,24 @@ private[hive] class HiveClientImpl(
       database: CatalogDatabase,
       ignoreIfExists: Boolean): Unit = withHiveState {
     val hiveDb = toHiveDatabase(database, Some(userName))
-    shim.createDatabase(client, hiveDb, ignoreIfExists)
+    try {
+      shim.createDatabase(client, hiveDb, ignoreIfExists)
+    } catch {
+      case _: AlreadyExistsException =>
+        throw new DatabaseAlreadyExistsException(database.name)
```

Comment:
I am wondering if we need to do something similar as:
https://github.com/apache/spark/blob/17aaf8ed29e24516ca460d92b481a0b2f0545ddf/sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveClientImpl.scala#L607-L612
@MaxGekk do you happen to know?